api/samlacs.go

```diff
@@ -237,7 +259,16 @@ func (a *API) SAMLACS(w http.ResponseWriter, r *http.Request) error {
 	}
 
 	var token *AccessTokenResponse
-
+	if err := db.Transaction(func(tx *storage.Connection) error {
+		if samlMetadataModified {
+			if terr := tx.Update(ssoProvider.SAMLProvider); terr != nil {
+				return terr
+			}
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
```

Comment:
```suggestion
	if samlMetadataModified {
		if err := db.Transaction(func(tx *storage.Connection) error {
			if terr := tx.Update(ssoProvider.SAMLProvider); terr != nil {
				return terr
			}
			return nil
		}); err != nil {
			return err
		}	
	}
```

Maybe something like this so a transaction is opened only if there should be a change?

Reply:
In fact, on second glance, you don't need the transaction you can just call Update on the DB object itself.